### PR TITLE
Add cluster spec doc for snow

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

https://github.com/aws/eks-anywhere/issues/1111

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/reference/clusterspec/snow.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+## SnowMachineConfig Fields
+
+### amiID (optional)
+AMI ID from which to create the machine instance. Snow provider offers an AMI lookup logic which will look for a suitable AMI ID based on the Kubernetes version and osFamily if the field is empty.
+
+### instanceType (optional)
+Type of the Snow EC2 machine instance. Permitted values: `sbe-c.large`, `sbe-c.xlarge`, `sbe-c.2xlarge`, `sbe-c.4xlarge` (Default: `sbe-c.large`).
+
+### osFamily
+Operating System on instance machines. Permitted values: bottlerocket, ubuntu.
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/snow.md:219]: Also, if bottlerocket is supported, do you just get it by default if you don't supply an ubuntu image?

File: docs/content/en/docs/reference/clusterspec/snow.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+### sshKeyName (optional)
+Name of the AWS Snow SSH key pair you want to configure to access your machine instances.
+
+The default is `eksa-default-{cluster-name}-{uuid}`.
+
+### devices
+A device IP list from which to bootstrap and provision machine instances.
+
+### network
+Custom network setting for the machine instances. Support DHCP and static IP configurations.
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/clusterspec/snow.md:233]:
```suggestion
Custom network setting for the machine instances. DHCP and static IP configurations are supported.
```

File: docs/content/en/docs/reference/clusterspec/snow.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+Refers to a `SnowIPPool` object which provides a range of ip addresses. When specified, an IP address selected from the pool will be allocated to the DNI.
+
+### containersVolume (optional)
+Configuration option for customizing containers data storage volume.
+
+### containersVolume.deviceName (optional)
+Name of the device for the containers data storage volume.
+
+### containersVolume.size
+Size of the storage device in Gi. Must be greater than the image snapshot size or 8 (whichever is greater).
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/clusterspec/snow.md:257]: Is "8" the default if it is not set?

**Review by chrisnegus — COMMENTED** [review]
Looks good. Just a few comments/questions.

File: docs/content/en/docs/reference/clusterspec/snow.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiayiwang7) [reply docs/content/en/docs/reference/clusterspec/snow.md:219]: yes bottlerocket is supported for snow. and yes the bottlerocket node image is preinstalled in the snow devices so if user selects bottlerocket as osFamily, the BR image in the device will be used

File: docs/content/en/docs/reference/clusterspec/snow.md (hunk 3 above)
Comment (jiayiwang7) [reply docs/content/en/docs/reference/clusterspec/snow.md:257]: no, size will be 0 if not set

Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/snow.md:257]: Okay. I'm confused about the 25 Gi below. Are you saying that if this is set at all, it must be at least 8Gi (presumably for Ubuntu OS), but must be at least 25 Gi for BottleRocket OS.

**Review by chrisnegus — COMMENTED** [review]
Looks good. I just had one question.

**Pull request merged**